pgosupport.cmake

```diff
@@ -0,0 +1,68 @@
+function(clr_pgo_unknown_arch)
+    if (WIN32)
+        message(FATAL_ERROR "Only AMD64, ARM and I386 are supported for PGO")
+    else()
+        message(FATAL_ERROR "PGO not currently supported on the current platform")
+    endif()
+endfunction(clr_pgo_unknown_arch)
+
+# Adds Profile Guided Optimization (PGO) flags to the current target
+function(add_pgo TargetName)
+    if(WIN32)
+        set(ProfileFileName "${TargetName}.pgd")
+    endif(WIN32)
+
+    file(TO_NATIVE_PATH
+        "$ENV{__PackagesDir}/Microsoft.DotNet.OptimizationData.Coreclr/$ENV{__BuildOS}.$ENV{__BuildArch}/${ProfileFileName}"
+        ProfilePath
+    )
+
+    # Enable PGO only for optimized configs
+    set(ConfigTypeList RELEASE RELWITHDEBINFO)
+
+    foreach(ConfigType IN LISTS ConfigTypeList)
```

Comment:
Ouch, that's unfortunate. I've already noticed that the generator expressions don't work at some places. So then please keep it the way you had it.